src/website/shared/models/linkage.py

```diff
@@ -62,6 +63,26 @@ def create_nixpkgs_issue(self) -> NixpkgsIssue:
         return issue
 
 
+class MaintainersEdit(models.Model):
+    """
+    A single manual edit of the list of maintainers of a suggestion.
+    """
+
+    class EditType(models.TextChoices):
+        ADD = "add", _("add")
+        REMOVE = "remove", _("remove")
+
+    edit_type = models.CharField(
+        max_length=text_length(EditType), choices=EditType.choices
+    )
+    maintainer = models.ForeignKey(NixMaintainer, on_delete=models.CASCADE)
+    suggestion = models.ForeignKey(
+        CVEDerivationClusterProposal,
+        related_name="maintainers_edits",
```

Comment:
Can we enforce in a constraint that a maintainer edit only appears once on a suggestion? We'd still need to ensure our UI logic won't run into the constraint, but at least the data model would make invalid state impossible to represent and we won't need to do post-checks.

Reply:
Do you mean that for example we don't have two edits that add the same maintainer to the same suggestion? Otherwise, reading things literally, I believe `ForeignKey` does enforce that an edit only belongs to one suggestion (it's a one-to-many relation).

Reply:
> Well, it's a one-to-many for edits, not for maintainers in those edits. You could have multiple edits with the same maintainer, right?

Yes, but this is ok... ? You might want to independently add maintainer `@tartempion` to suggestion A and remove it from suggestion B. You initially wrote 

> Can we enforce in a constraint that a maintainer edit only appears once on a suggestion

which I believe is indeed enforced by the foreign key. So, if I understand correctly, the only remaining constraint that isn't enforced and that we want to enforce is the one you said, that several edits to the _same package_ with the _same maintainer_ shouldn't be allowed. Is that correct?

Reply:
> Is that correct?

Yes, that's what I meant.